lib/mayaUsd/fileio/primWriterRegistry.cpp

```diff
@@ -74,32 +157,57 @@ void UsdMayaPrimWriterRegistry::RegisterRaw(
 }
 
 /* static */
-UsdMayaPrimWriterRegistry::WriterFactoryFn
-UsdMayaPrimWriterRegistry::Find(const std::string& mayaTypeName)
+UsdMayaPrimWriterRegistry::WriterFactoryFn UsdMayaPrimWriterRegistry::Find(
+    const std::string&          mayaTypeName,
+    const UsdMayaJobExportArgs& exportArgs,
+    const MObject&              exportObj)
 {
     TfRegistryManager::GetInstance().SubscribeTo<UsdMayaPrimWriterRegistry>();
 
-    // unfortunately, usdTypeName is diff from the tfTypeName which we use to
-    // register.  do the conversion here.
-    WriterFactoryFn ret = nullptr;
-    if (TfMapLookup(_reg, mayaTypeName, &ret)) {
-        return ret;
+    _Registry::const_iterator it = _Find(mayaTypeName, exportArgs, exportObj);
+
+    if (it != _reg.end()) {
+        return it->second._writer;
     }
 
     static const TfTokenVector SCOPE = { _tokens->UsdMaya, _tokens->PrimWriter };
     UsdMaya_RegistryHelper::FindAndLoadMayaPlug(SCOPE, mayaTypeName);
 
+    it = _Find(mayaTypeName, exportArgs, exportObj);
+
+    if (it != _reg.end()) {
+        return it->second._writer;
+    }
+
     // ideally something just registered itself.  if not, we at least put it in
     // the registry in case we encounter it again.
```

Comment:
This comment should be moved a few lines up before the 2nd _Find.

Reply:
This comment was trying to explain what the code below does so I'll reword it to make it clearer